### PR TITLE
Corrects use of snow flags and adds flag consistency checks

### PR DESCRIPTION
**Pull request by njeffery**
-Config_use_effective_density activates density from compaction only which is needed for ITD snow redistribution
-Config_use_snow_grain_radius activates snow grain radius and also needs snow liquid and snow ice tracers
-Includes a bugfix in snow_redistribution when using 1 snow layer. Will need to be fixed in Icepack

BFB with snow flag defaults
except for the very initial value of effective snow density

### CODE REVIEW

File: components/mpas-seaice/src/shared/mpas_seaice_column.F [hunk 1]
```diff
@@ -10141,6 +10149,35 @@ subroutine check_column_package_configs(domain)
             messageType=MPAS_LOG_CRIT)
     endif
 
+    ! check config_snow_redistribution_scheme
+      if ((trim(config_snow_redistribution_scheme) == "ITDrdg" .or. &
+         trim(config_snow_redistribution_scheme) == "ITDsd") .and. .not. config_use_effective_snow_density) then
```
Comment (eclare108213) [inline components/mpas-seaice/src/shared/mpas_seaice_column.F:10154]: should these configuration flag checks also be in mpas_seaice_icepack.F?

File: components/mpas-seaice/src/column/ice_snow.F90 [hunk 2]
```diff
@@ -422,6 +412,8 @@ subroutine snow_redist(dt, nslyr, ncat, wind, ain, vin, vsn, zqsn, &
                                         zs1(:),   zs2(:),     &
                                         hslyr,    hsn_new(n), &
                                         zqsn(:,n))
+               else
+                  hsn_new(1) = hsn_new(1) + dhsn
```
Comment (eclare108213) [inline components/mpas-seaice/src/column/ice_snow.F90:416]: this is the bug fix

**Review by eclare108213 — COMMENTED** [review]
It looks like most of the changes here are just moving things around so that the ice and liquid mass content is associated with the snow grain radius (metamorphosis) processes/tracers rather than wind redistribution, logically.  The bug fix in snow_redist won't affect runs with more than one snow layer.  Has this set of changes been tested in D cases yet?  (Will 'column' run?  I assume that 'icepack' won't run, yet.)

**Comment by eclare108213** [comment]
I'll set up D-case regression tests for 'column'.

File: components/mpas-seaice/src/shared/mpas_seaice_column.F (hunk 1 above)
Comment (njeffery) [reply components/mpas-seaice/src/shared/mpas_seaice_column.F:10154]: Yes.  I'll add those.

File: components/mpas-seaice/src/column/ice_snow.F90 (hunk 2 above)
Comment (njeffery) [reply components/mpas-seaice/src/column/ice_snow.F90:416]: yes

**Comment by eclare108213** [comment]
Confirmed that 3-month 'column' D-cases with all of the snow flags turned on are BFB before and after this PR.  Also confirmed that both 'column' and 'icepack' 3-month D-cases are BFB before and after this PR with the snow flags turned off.

**Review by eclare108213 — APPROVED** [review]

**Review by proteanplanet — APPROVED** [review]
Since this successfully passed a D-case test, I approve

**Pull request merged**